packages/lexical-markdown/src/MarkdownTransformers.ts

```diff
@@ -153,6 +153,17 @@ export type TextMatchTransformer = Readonly<{
   type: 'text-match';
 }>;
 
+const ORDERED_LIST_REGEX = /^(\s*)(\d{1,})\.\s/;
+const UNORDERED_LIST_REGEX = /^(\s*)[-*+]\s/;
+const CHECK_LIST_REGEX = /^(\s*)(?:-\s)?\s?(\[(\s|x)?\])\s/i;
+const HEADING_REGEX = /^(#{1,6})\s/;
+const QUOTE_REGEX = /^>\s/;
+const CODE_START_REGEX = /^[ \t]*```(\w+)?/;
+const CODE_END_REGEX = /[ \t]*```$/;
+const CODE_SINGLE_LINE_REGEX = /^[ \t]*```(\w+)?[^`]*```/;
```

Comment:
I don't think this regex is correct, it doesn't handle embedded backticks and the `\w+` is not relevant to inline code. Maybe something like this would work?
```suggestion
const CODE_SINGLE_LINE_REGEX = /^[ \t]*```(?:[^`]+|``?[^`])+```(?:[^`]|$)/;
```